test-framework/org.jboss.tools.openshift.reddeer/src/org/jboss/tools/openshift/reddeer/view/resources/OpenShiftODOConnection.java

```diff
@@ -0,0 +1,143 @@
+/*******************************************************************************
+ * Copyright (c) 2020 Red Hat, Inc.
+ * Distributed under license by Red Hat, Inc. All rights reserved.
+ * This program is made available under the terms of the
+ * Eclipse Public License v2.0 which accompanies this distribution,
+ * and is available at http://www.eclipse.org/legal/epl-v20.html
+ *
+ * Contributors:
+ * Red Hat, Inc. - initial API and implementation
+ ******************************************************************************/
+package org.jboss.tools.openshift.reddeer.view.resources;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import org.eclipse.reddeer.common.exception.RedDeerException;
+import org.eclipse.reddeer.common.wait.TimePeriod;
+import org.eclipse.reddeer.common.wait.WaitWhile;
+import org.eclipse.reddeer.swt.api.TreeItem;
+import org.eclipse.reddeer.swt.condition.ShellIsAvailable;
+import org.eclipse.reddeer.swt.impl.menu.ContextMenuItem;
+import org.eclipse.reddeer.workbench.core.condition.JobIsRunning;
+import org.jboss.tools.openshift.reddeer.dialogs.InputDialog;
+import org.jboss.tools.openshift.reddeer.utils.DatastoreOS3;
+import org.jboss.tools.openshift.reddeer.utils.OpenShiftLabel;
+import org.jboss.tools.openshift.reddeer.view.OpenShiftApplicationExplorerView;
+
+/**
+ * 
+ * OpenShift Application Explorer Connection implemented with RedDeer.
+ * 
+ * @author jkopriva@redhat.com
+ *
+ */
+public class OpenShiftODOConnection extends AbstractOpenShiftApplicationConnection {
+	
+	public OpenShiftODOConnection(TreeItem connectionItem) {
+		super(connectionItem);
+	}
+
+	/**
+	 * Gets an OpenShift project with specified project name.
+	 *  
+	 * @param projectName name of a project, if displayed name is provided,
+	 *  	its displayed name, otherwise its project name
+	 * @return OpenShift project
+	 */
+	public OpenShiftODOProject getProject(String projectName) {
+		activateOpenShiftApplicationExplorerView();
+		item.expand();
+		
+		new WaitWhile(new JobIsRunning(), TimePeriod.VERY_LONG);
```

Comment:
VERY_LONG timeout for item expand? Isn't that quite a bit? Also, if there is specific job, use its name to wait for.

Reply:
changed to LONG